### PR TITLE
Block Validation: Allow unitless zero CSS lengths

### LINKED ISSUE #28497: Cover: Block may invalidate with optimized styles
## Description

The Cover block with a focal point will generate some HTML like the following:

```html
<img class="wp-block-cover__image-background wp-image-703" alt="" src="https://example.com/image.jpeg" style="object-position:0% 44%" data-object-fit="cover" data-object-position="0% 44%"/>
```

When a simple CSS optimization plugin runs on post save (for example, applying [this](https://packagist.org/packages/cerdic/css-tidy) to style attributes), it removes the redundant `%` from `0%`. The following post content is stored to the database:

```html
<img class="wp-block-cover__image-background wp-image-703" alt="" src="https://example.com/image.jpeg" style="object-position:0 44%;" data-object-fit="cover" data-object-position="0% 44%" />
```

When the editor is refreshed, the post content is invalid. The CSS is semantically the same, but the missing redundant `%` causes block invalidation to occur. The block should be resilient to this change as the style is semantically identical.

Note that https://github.com/WordPress/gutenberg/issues/28487 makes this issue much more likely to be encountered as a `0` value must be picked in the focal point picker.

## Step-by-step reproduction instructions

You need to run a css optimization plugin that applies the `%` removal optimization on post save.

- Add a Cover block
- Add a background image
- Pick a focal point in the block sidebar
- Save the post
- Refresh the editor
- Notice the invalidation (only if a plugin is optimizing the style attribute).

## Expected behaviour

There should be no block invalidation.

## Actual behaviour

The block is invalidated.

## WordPress information
- WordPress version: 5.6
- Gutenberg version: 9.8.1
- Are all plugins except Gutenberg deactivated? **No**
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

### PR DESCRIPTION
**Pull request by ockham**
## Description
Fixes #28497. Quoting that issue:

> The Cover block with a focal point will generate some HTML like the following:
> 
> ```
> 
> ```
> 
> When a simple CSS optimization plugin runs on post save (for example, applying [this](https://packagist.org/packages/cerdic/css-tidy) to style attributes), it removes the redundant `%` from `0%`. The following post content is stored to the database:
> 
> ```
> 
> ```
> 
> When the editor is refreshed, the post content is invalid. The CSS is semantically the same, but the missing redundant `%` causes block invalidation to occur. The block should be resilient to this change as the style is semantically identical.
> 
> Note that #28487 makes this issue much more likely to be encountered as a `0` value must be picked in the focal point picker.

This PR changes `style` attribute normalization (which is part of block validation) accordingly.

## How has this been tested?

```
npm run test-unit packages/blocks/src/api/test/validation.js
```

Furthermore, verify that #28497 is fixed (see that issue's instructions to reproduce).

### CODE REVIEW

File: packages/blocks/src/api/validation/index.js [hunk 1]
```diff
@@ -325,6 +332,21 @@ export function isEquivalentTextTokens(
...
+ *
+ * @param {string} value CSS length value.
+ *
+ * @return {string} Normalized CSS length value.
+ */
+export function getNormalizedLength( value ) {
+	if ( CSS_UNITS.some( ( unit ) => value === '0' + unit ) ) {
+		return '0';
+	}
+	return value;
```
Comment (aristath) [inline packages/blocks/src/api/validation/index.js:347]: Why would we limit this to specific units? Would it not be the same if we did something like this?
```js
return 0 === parseFloat( value ) ? 0 : value;
```
`0pt`, `0ch` and all other units are also valid, wouldn't using `parseFloat` be safer and future-proof?

Comment (ockham) [reply packages/blocks/src/api/validation/index.js:347]: I wasn't aware that [`parseFloat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) cuts off trailing strings :grimacing: 

Anyway, thanks for your suggestion, makes sense to me. I'll change it accordingly!

**Review by sirreal — APPROVED** [review]
This fixed the validation issues in that case described in #28497 in my testing. Looks good 👍

**Comment by ockham** [comment]
Some media modal related e2e tests are failing because of a known regression in Core: https://core.trac.wordpress.org/ticket/50025

**Comment by ockham** [comment]
Cherry-picked to `release/9.8`.

**Pull request merged**